lib/foodcritic/api.rb

```diff
@@ -77,6 +77,20 @@ def chef_solo_search_supported?(recipe_path)
       end
     end
 
+    # The absolute path of a cookbook from the specified file.
+    def cookbook_base_path(file)
+      file = File.expand_path(file) # make sure we get an absolute path
+      file = File.dirname(file) unless File.directory?(file) # get the dir only
+
+      # get list of items in the dir and intersect with metadata array.
+      # until we get an interfact (we have a metadata) walk up the dir structure
+      until (Dir.entries(file) & %w(metadata.rb metadata.json)).any?
+        file = File.absolute_path(File.dirname(file.to_s))
```

Comment:
I think we can skip the `File.absolute_path` since we've run `File.expand_path` on its descendant